### PR TITLE
DAT-3711 Adding sections to the infobox builder

### PR DESCRIPTION
**Pull request by sebastianmarzjan**
## Links
- http://wikia-inc.atlassian.net/browse/DAT-3711
## Description

This PR introduces serialization and deserialization on the app side needed by the infobox builder after introducing section headers.

### CODE REVIEW

File: extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php [hunk 1]
```diff
@@ -160,4 +149,119 @@ protected function addNode( $node, SimpleXMLElement $xml ) {
...
+		$infoboxDom = $this->createInfoboxDom( $formatted );
+
+		$inGroup = false;
+		$currentGroupDom = null;
+
+		foreach ( $xml->children() as $currentChildNode ) {
+			$childNodeDom = dom_import_simplexml( $currentChildNode );
+
+			if ( !$inGroup ) {
+				if ( $currentChildNode->getName() !== 'header' ) {
```
Comment (dianafa) [inline extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php:194]: can we check here `$currentChildNode->getName() === 'header'` ?

File: extensions/wikia/PortableInfoboxBuilder/tests/PortableInfoboxBuilderServiceTest.php [hunk 2]
```diff
@@ -39,14 +39,14 @@ public function testMarkupSupport( $markup, $expected ) {
...
-	public function testUpdateInfobox($data, $expected) {
+	public function tzestUpdateInfobox($data, $expected) {
 		$this->assertEquals( $expected, $this->builderService->updateInfobox($data['oldInfobox'], $data['newInfobox'], $data['oldContent']));
 	}
 
 	/**
 	 * @dataProvider updateDocumentationProvider
 	 */
-	public function testUpdateDocumentation($data, $expected) {
+	public function tzŃestUpdateDocumentation($data, $expected) {
```
Comment (dianafa) [inline extensions/wikia/PortableInfoboxBuilder/tests/PortableInfoboxBuilderServiceTest.php]: : o

File: extensions/wikia/PortableInfoboxBuilder/tests/PortableInfoboxBuilderServiceTest.php [hunk 3]
```diff
@@ -39,14 +39,14 @@ public function testMarkupSupport( $markup, $expected ) {
 	/**
 	 * @dataProvider updateInfoboxProvider
 	 */
-	public function testUpdateInfobox($data, $expected) {
+	public function tzestUpdateInfobox($data, $expected) {
```
Comment (dianafa) [inline extensions/wikia/PortableInfoboxBuilder/tests/PortableInfoboxBuilderServiceTest.php]: : o

File: extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php (hunk 1 above)
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php]: thanks, going to fix right away

File: extensions/wikia/PortableInfoboxBuilder/tests/PortableInfoboxBuilderServiceTest.php (hunk 2 above)
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/tests/PortableInfoboxBuilderServiceTest.php]: hmm, strange things there. Going to fix :)

File: extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php (hunk 1 above)
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php:194]: This is the default, most typical (in a normal infobox) case. This is why I'd like it to be the first one. Any strong counter-opinion ?

Comment (dianafa) [inline extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php:194]: it's just easier for a human to process this logic without negation. And === looks nicer than !==. Not very strong though.

**Pull request merged**